src/lib/PnP.Framework/AuthenticationManager.cs

```diff
@@ -867,7 +869,8 @@ public async Task<ClientContext> GetContextAsync(string siteUrl, CancellationTok
                     {
                         var context = GetAccessTokenContext(siteUrl, (site) =>
                         {
-                            return this.acsTokenGenerator.GetToken(new Uri(site));
+                            // HACK: sync over async

```

Comment:
@fowl2 : what's the advantage here?

Reply:
it's incrementally converting the code to be async. This is equivalent to what the other auth methods are doing in `BuildClientContext`.  Perhaps that was just because MSAL has no sync apis(?) 

Presumably an more async-friendly alternative to `ExecutingWebRequest` will be added to the CSOM at some point and this can all be made async all the way through.  Maybe it'd be better to hold off on this part.